upgrade/upgrade.go

```diff
@@ -18,25 +18,82 @@ package upgrade
 
 import (
 	"context"
+	"fmt"
+	"sort"
 
 	"cosmossdk.io/log"
 	upgradetypes "cosmossdk.io/x/upgrade/types"
+	ismkeeper "github.com/bcp-innovations/hyperlane-cosmos/x/core/01_interchain_security/keeper"
+	ismtypes "github.com/bcp-innovations/hyperlane-cosmos/x/core/01_interchain_security/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/module"
+	authoritytypes "github.com/noble-assets/authority/types"
 )
 
 func CreateUpgradeHandler(
 	mm *module.Manager,
 	cfg module.Configurator,
 	logger log.Logger,
+	ismKeeper ismkeeper.Keeper,
 ) upgradetypes.UpgradeHandler {
 	return func(ctx context.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		sdkCtx := sdk.UnwrapSDKContext(ctx)
+
 		vm, err := mm.RunMigrations(ctx, cfg, vm)
 		if err != nil {
 			return vm, err
 		}
 
+		// Because the HyperEVM default ISM has already been created on Noble's
+		// testnet, we only need to run the logic for creating the HyperEVM
+		// default ISM on Noble's mainnet.
+		if sdkCtx.ChainID() == MainnetChainID {
+			err = configureHyperlaneISM(ctx, logger, ismKeeper)
+			if err != nil {
+				return vm, err
+			}
+		}
+
 		logger.Info(UpgradeASCII)
 
 		return vm, nil
 	}
 }
+
+func configureHyperlaneISM(ctx context.Context, logger log.Logger, keeper ismkeeper.Keeper) error {
+	// https://docs.hyperlane.xyz/docs/reference/addresses/validators/mainnet-default-ism-validators#hyperevm-999
+	domain := uint32(999)
```

Comment:
xs nit: `domain` could be a `const` instead of a `var`

Reply:
https://github.com/noble-assets/noble/pull/607/commits/47bd42392d33d84298c81c5c5c2bb39e020770b7